Northstar.CustomServers/mod/scripts/vscripts/mp/_base_gametype.gnut

```diff
@@ -1412,6 +1412,32 @@ void function CodeCallback_WeaponFireInCloak( entity player )
 		//player.SetCloakFlicker( 1.0, 2.0 )
 
 		DisableCloak( player, 0.5 )
+		entity cloakWeapon
+		int offhandSlot
+		for ( int i = 0; i <= OFFHAND_MELEE; i++ ) // OFFHAND_MELEE is the largest
+		{
+			entity nowWeapon = player.GetOffhandWeapon( i )
+			if( IsValid( nowWeapon ))
+			{
+				if( nowWeapon.GetWeaponClassName() == "mp_ability_cloak" )
+				{
+					cloakWeapon = nowWeapon
+					offhandSlot = i
+				}
+			}
+		}
+		if( IsValid( cloakWeapon ) )
+		{
+			array<string> mods = cloakWeapon.GetMods()
+			player.TakeWeapon( "mp_ability_cloak" ) // not using TakeWeaponNow() to fit vanilla behavior
+			player.GiveOffhandWeapon( "mp_ability_cloak", offhandSlot, mods )
+			cloakWeapon = player.GetOffhandWeapon( offhandSlot )
+			cloakWeapon.SetWeaponPrimaryClipCountAbsolute( 0 )
+		}
+
+		//printt( "weapon", weapon.GetWeaponClassName() )
```

Comment:
probably not worth keeping the old code in a comment, instead maybe have a comment explaining that you can't reset cooldown properly in script, and none of the respawn code that was there originally